libs/messaging/messaging-impl/src/main/kotlin/net/corda/messaging/mediator/RPCClient.kt

```diff
@@ -47,14 +47,19 @@ class RPCClient(
         }
     }
 
-    private fun processMessage(message: MediatorMessage<*>): MediatorMessage<*> {
+    private fun processMessage(message: MediatorMessage<*>): MediatorMessage<*>? {
         val request = buildHttpRequest(message)
         val response = sendWithRetry(request)
 
         checkResponseStatus(response.statusCode())
 
         val deserializedResponse = deserializePayload(response.body())
-        return MediatorMessage(deserializedResponse, mutableMapOf("statusCode" to response.statusCode()))
+
+        return if(deserializedResponse==null){
```

Comment:
```suggestion
        return deserializedResponse?.let {
            MediatorMessage(deserializedResponse, mutableMapOf("statusCode" to response.statusCode()))
        }
```
Nitpick 'cause I like Kotlin's idiomatic null-checking, but if you think yours is more explicit feel free to keep it.